gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskType.java

```diff
@@ -0,0 +1,247 @@
+        }
+      };
+
+  private static final long serialVersionUID = -944042261695072026L;
+
+  private final BigInteger id;
+  private final DiskTypeId diskTypeId;
+  private final String creationTimestamp;
+  private final String description;
+  private final String validDiskSize;
+  private final String selfLink;
+  private final Long defaultDiskSizeGb;
+  private final DeprecationStatus<DiskTypeId> deprecationStatus;
+
+  static final class Builder {
+
+    private BigInteger id;
+    private DiskTypeId diskTypeId;
+    private String creationTimestamp;
+    private String description;
+    private String validDiskSize;
+    private String selfLink;
+    private Long defaultDiskSizeGb;
+    private DeprecationStatus<DiskTypeId> deprecationStatus;
+
+    private Builder() {}
+
+    Builder id(BigInteger id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder creationTimestamp(String creationTimestamp) {
+      this.creationTimestamp = creationTimestamp;
+      return this;
+    }
+
+    Builder diskTypeId(DiskTypeId diskTypeId) {
+      this.diskTypeId = diskTypeId;
+      return this;
+    }
+
+    Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    Builder validDiskSize(String validDiskSize) {
+      this.validDiskSize = validDiskSize;
+      return this;
+    }
+
+    Builder selfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    Builder defaultDiskSizeGb(Long defaultDiskSizeGb) {
+      this.defaultDiskSizeGb = defaultDiskSizeGb;
+      return this;
+    }
+
+    Builder deprecationStatus(DeprecationStatus<DiskTypeId> deprecationStatus) {
+      this.deprecationStatus = deprecationStatus;
+      return this;
+    }
+
+    DiskType build() {
+      return new DiskType(this);
+    }
+  }
+
+  private DiskType(Builder builder) {
+    this.id = builder.id;
+    this.creationTimestamp = builder.creationTimestamp;
+    this.diskTypeId = builder.diskTypeId;
+    this.description = builder.description;
+    this.validDiskSize = builder.validDiskSize;
+    this.selfLink = builder.selfLink;
+    this.defaultDiskSizeGb = builder.defaultDiskSizeGb;
+    this.deprecationStatus = builder.deprecationStatus;
+  }
+
+  /**
+   * Returns the creation timestamp in RFC3339 text format.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
+   */
+  public String creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the disk type's identity.
+   */
+  public DiskTypeId diskTypeId() {
+    return diskTypeId;
+  }
+
+  /**
+   * Returns an unique identifier for the disk type; defined by the service.
+   */
+  public BigInteger id() {
+    return id;
+  }
+
+  /**
+   * Returns a textual description of the disk type.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns an optional textual description of the valid disk size, such as "10GB-10TB".
```

Comment:
Yes we can parse it even though there's no real documentation of which units if measure can actually be returned (mb, gb, tb). Do you think it's safe enough for us to parse it and assume that possible units are only the tree above?